test/conformance/conformance_arena_constraints.cpp

```diff
@@ -75,97 +75,132 @@ TEST_CASE("Test core types topology traversal correctness") {
-    auto constraint_concurrency = oneapi::tbb::info::default_concurrency(c);
-    REQUIRE_MESSAGE(constraint_concurrency == numa_task_arenas[numa_i].max_concurrency(),
-        "Maximum concurrency level of task_arena should be the same as for constraints");
-  }
+    system_info::initialize();
+    auto numa_indices = oneapi::tbb::info::numa_nodes();
+    using return_type = decltype(oneapi::tbb::create_numa_task_arenas());
+    static_assert(
+        std::is_same<std::vector<oneapi::tbb::task_arena>, return_type>::value,
+        "Return type of oneapi::tbb::create_numa_task_arenas() does not match the type guaranteed by oneAPI Specification"
+    );
+    return_type numa_task_arenas = oneapi::tbb::create_numa_task_arenas();
+
+    REQUIRE_MESSAGE(numa_task_arenas.size() == numa_indices.size(),
+        "create_numa_task_arenas returns the same number of NUMA nodes as tbb::info::numa_nodes()");
+    // Test that arenas are not initialized
+    for (auto& ta : numa_task_arenas) {
+        REQUIRE_MESSAGE(!ta.is_active(),
+            "create_numa_task_arenas must return a vector of non-initialized arenas");
+    }
+
+    for (std::size_t numa_i = 0; numa_i < numa_indices.size(); ++numa_i) {
+        oneapi::tbb::task_arena::constraints c{numa_indices[numa_i]};
+        auto constraint_concurrency = oneapi::tbb::info::default_concurrency(c);
+        REQUIRE_MESSAGE(constraint_concurrency == numa_task_arenas[numa_i].max_concurrency(),
+            "Maximum concurrency level of task_arena should be the same as for constraints");
+    }
 }
 
 struct join_arena_observer : tbb::task_scheduler_observer {
-  join_arena_observer(tbb::task_arena &ta, int max_workers, int max_external_threads)
-      : tbb::task_scheduler_observer(ta), max_num_workers(max_workers), max_num_external_threads(max_external_threads)
-  {
-    observe(true);
-  }
-
-  void on_scheduler_entry(bool is_worker) override {
-    int prev;
-    if (is_worker) {
-      prev = num_workers.fetch_add(1, std::memory_order_relaxed);
-      REQUIRE_MESSAGE(prev+1 <= max_num_workers, "More than expected worker threads has joined arena");
-    } else {
-      prev = num_external_threads.fetch_add(1, std::memory_order_relaxed);
-      REQUIRE_MESSAGE(prev+1 <= max_num_external_threads, "More than expected external threads has joined arena");
+    join_arena_observer(tbb::task_arena &ta, int max_workers, int max_external_threads)
+        : tbb::task_scheduler_observer(ta), max_num_workers(max_workers), max_num_external_threads(max_external_threads)
+    {
+        observe(true);
     }
-  }
 
-  int max_num_workers;
-  int max_num_external_threads;
-  std::atomic_int num_workers{};
-  std::atomic_int num_external_threads{};
+    void on_scheduler_entry(bool is_worker) override {
+        int current;
+        int expected_peak;
+        if (is_worker) {
+            current = num_workers.fetch_add(1, std::memory_order_relaxed) + 1;
```

Comment:
Let's put a TODO comment here with text something like:
```suggestion
            // TODO: Adapt utils::ConcurrencyTracker for its reuse here and in the else branch below
            current = num_workers.fetch_add(1, std::memory_order_relaxed) + 1;
```